### ISSUE TITLE
how to use latest android-ndk-rs

### ISSUE THREAD

**Issue by nikita-skobov** [not-fl3/miniquad#185]
This issue is related to the fork of android-ndk-glue located here (https://github.com/not-fl3/android-rs-glue) but I am opening this issue on miniquad's repo because issues arent available on that fork.

This fork uses specific pinned versions of dependencies that seem to break in later versions of rust.
I was wondering how much difficulty would it be to modify that fork to use the latest version of the android-ndk-rs project (https://github.com/rust-windowing/android-ndk-rs).

The specific issue im currently running into can be reproduced by

1. `git clone https://github.com/not-fl3/android-rs-glue && cd android-rs-glue`
2. Edit the Dockerfile to use a later version of rust: (add the following lines)
```
FROM rust:stretch

RUN apt-get update
RUN apt-get install -yq openjdk-8-jre unzip wget cmake

# add these lines:
RUN rustup toolchain install 1.50.0
RUN rustup default 1.50.0
# keep the rest the same
```
3. build the docker image `docker build -t notfl3gluetest .`
4. in another directory, clone miniquad: `git clone https://github.com/not-fl3/miniquad && cd miniquad`
5. Run the docker image interactively: `docker run -it --rm -v $(pwd):/root/src -w /root/src notfl3gluetest`
6. When inside the docker image, build using cargo apk: `cargo apk build --example quad`

Build fails with:

```
thread 'main' panicked at 'attempted to zero-initialize type `sized_chunks::sized_chunk::Chunk<(core::package_id::PackageId, im_rc::OrdMap<core::package_id::PackageId, std::vec::Vec<core::dependency::Dependency>>)>`, which is invalid', /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/mem/mod.rs:623:9
```

I think the reason is that the not-fl3/android-rs-glue repo has some pinned dependencies which cause conflicts when trying to use more recent rust versions.

# Why is this a problem:

The existing `notfl3/cargo-apk` docker image comes with rustc version 1.47.0 which works fine and can build the miniquad examples, however I am trying to make a project that uses other dependencies, and some of those other dependencies depend on features of rust that are only available in later versions (ie 1.50.0).

Additionally, I was wondering if it would be possible to change the dockerfile to use the current `android-ndk-rs` project instead of the fork because then it is more likely to be maintained and up to date. I know that in your fork you changed some of the actual glue code: (https://github.com/rust-windowing/android-rs-glue/commit/cf76e53e96e4d08e25fd90d609ad2a9e94e732f6) but I was wondering it theres a way for miniquad to use its sokol headers glue code without modifying android-ndk-rs?

Id be willing to help with this if you can point me in the right direction.

**Comment by nikita-skobov** [comment]
I actually went ahead and forked android-rs-glue again and modified their procedural macros to be compatible with miniquad. I tested it and now it works with the dependency that I was trying to use. see it here: https://github.com/nikita-skobov/android-ndk-rs/commit/9cc4af43756d13d3940dd640a01e35ed994b8175#diff-97ba036ced5dadf173f84ad3cbf9db153d947f441cbc59a7bb77858a784c802bR112

This fork depends on pull request #186 because sapp_android needs to be public for the procedural macro to work.
